xxh_aarch64dispatch.S

```diff
@@ -0,0 +1,1043 @@
+#include "asmdefs.h"
+
+.data
+
+xxh3_prime_1:	.dword	0x9E3779B1U
+
+.text
+
+.macro init_index
+	mov	z1.d, #2
+	pfalse	p1.b
+	index	z7.d, #1, #1
+	trn1	p1.d, p1.d, p7.d
+	sub	z7.d, p1/m, z7.d, z1.d
+.endm
+
+.macro calc_acc512 acc
+	// swapped = SWAP(input)
+	tbl	z5.d, {z1.d}, z7.d
+	// mixed = input EOR secret
+	eor	z1.d, p7/m, z1.d, z2.d
+	mov	z6.d, z1.d
+	// mixed_lo = mixed AND 0xffffffff
+	and	z1.d, p7/m, z1.d, z3.d
+	// mixed_hi = mixed >> 32
+	lsr	z6.d, p7/m, z6.d, z4.d
+	// mixed_lo = mixed_hi * acc + mixed_lo
+	mad	z1.d, p7/m, z6.d, \acc
+	// acc = mixed_lo + swapped
+	add	z1.d, p7/m, z1.d, z5.d
+.endm
+
+.macro calc_scram acc
+	eor	z1.d, \acc, z1.d
+	lsr	z5.d, \acc, #47
+	eor	\acc, z1.d, z5.d
+	mul	\acc, p7/m, \acc, z8.d
+.endm
+
+/*
+ * Input Registers: x0 (acc), x1 (nbStripeSoFarPtr), x2 (nbStripesPerBlock),
+ *                  x3 (input), x4 (nbStripes), x5 (secret), x6 (secretLimit)
+ * Output Registers: None
+ * Modified Registers: p1, p7, z0-z12, x1-x4, x6-x17
+ */
+ENTRY (XXH3_aarch64_sve128_consume_stripes)
+	stp	x29, x30, [sp, #-16]!
+	addvl	sp, sp, #-6
+	mov	x29, sp
+	/*
+	 * prepare index
+	 * p7 and z7 are reserved to use.
+	 */
+	ptrue	p7.d
+	init_index
+
+	st1d	{z8.d}, p7, [sp, #1, MUL VL]	// save z8
```

Comment:
If I'm not mistaken, z16-z31 are caller saved.

Reply:
It's mentioned in AAPCS64.

`z0-z7 are used to pass scalable vector arguments to a subroutine, and to return scalable vector results from a function. If a subroutine takes at least one argument in scalable vector registers or scalable predicate registers, or if it is a function that returns results in such registers, it must ensure that the entire contents of z8-z23 are preserved across the call. In other cases it need only preserve the low 64 bits of z8-z15, as described in SIMD and Floating-Point registers.`

So I have to save z8. And I also tried in program, I may meet some software error without saving z8.

Reply:
I can try to use z16. I don't know z16 is saved by caller before, and I couldn't find it from AAPCS64. I'll try it.